src/main/java/com/gregtechceu/gtceu/api/mui/base/drawable/IKey.java

```diff
@@ -0,0 +1,270 @@
+package com.gregtechceu.gtceu.api.mui.base.drawable;
+
+import com.google.gson.JsonObject;
+import com.gregtechceu.gtceu.api.mui.base.IJsonSerializable;
+import com.gregtechceu.gtceu.api.mui.drawable.text.*;
+import com.gregtechceu.gtceu.api.mui.theme.WidgetTheme;
+import com.gregtechceu.gtceu.api.mui.utils.Alignment;
+import com.gregtechceu.gtceu.api.mui.widgets.TextWidget;
+import com.gregtechceu.gtceu.client.mui.screen.viewport.GuiContext;
+import com.gregtechceu.gtceu.utils.serialization.json.JsonHelper;
+import net.minecraft.ChatFormatting;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraftforge.api.distmarker.Dist;
+import net.minecraftforge.api.distmarker.OnlyIn;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.function.Supplier;
+
+/**
+ * This represents a piece of text in a GUI.
+ */
+public interface IKey extends IDrawable, IJsonSerializable<IKey> {
```

Comment:
The name IKey doesn't clearly indicate that it's a text element, also, this might be better as a class rather than interface